src/platforms/ios/iosiaphandler.mm

```diff
@@ -214,6 +237,9 @@ - (void)paymentQueueRestoreCompletedTransactionsFinished:(nonnull SKPaymentQueue
 
   [delegate dealloc];
   m_delegate = nullptr;
+  if (@available(iOS 15, *)) {
+    swiftIAPHandler = nullptr;
```

Comment:
Do we need to invoke a destructor or `free()` kind of method here, or does Objective-C do that automatically?

Reply:
Once there is no reference to the object, it will automatically be destroyed.